pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,324 @@
+				Namespace: render.ElasticsearchNamespace,
+			},
+			Data: map[string][]byte{
+				"username": []byte("elastic"),
+				"password": esAdminUserSecret.Data["elastic"],
+			},
+		}
+		secrets = append(secrets, elasticsearchUserSecret)
+		tlsAnnotations[render.ElasticsearchTLSHashAnnotation] = rmeta.SecretsAnnotationHash(append(certSecrets, esAdminUserSecret, esInternalCertSecret)...)
+	} else {
+		tlsAnnotations[render.ElasticsearchTLSHashAnnotation] = rmeta.SecretsAnnotationHash(append(certSecrets, esInternalCertSecret)...)
+	}
+
+	// tigera-secure-kb-http-certs-public, mounted by ES Gateway.
+	secrets = append(secrets, secret.CopyToNamespace(render.ElasticsearchNamespace, kibanaInternalCertSecret)...)
+	tlsAnnotations[render.KibanaTLSAnnotationHash] = rmeta.SecretsAnnotationHash(kibanaInternalCertSecret)
+	return &esGateway{
+		logStorage:     logStorage,
+		installation:   installation,
+		pullSecrets:    pullSecrets,
+		secrets:        secrets,
+		tlsAnnotations: tlsAnnotations,
+		clusterDomain:  clusterDomain,
+	}
+}
+
+type esGateway struct {
+	logStorage     *operatorv1.LogStorage
+	installation   *operatorv1.InstallationSpec
+	pullSecrets    []*corev1.Secret
+	secrets        []*corev1.Secret
+	tlsAnnotations map[string]string
+	clusterDomain  string
+	csrImage       string
+	esGatewayImage string
+}
+
+func (e *esGateway) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := e.installation.Registry
+	path := e.installation.ImagePath
+	prefix := e.installation.ImagePrefix
+	var err error
+	errMsgs := []string{}
+
+	e.esGatewayImage, err = components.GetReference(components.ComponentESGateway, reg, path, prefix, is)
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+	if e.installation.CertificateManagement != nil {
+		e.csrImage, err = render.ResolveCSRInitImage(e.installation, is)
+		if err != nil {
+			errMsgs = append(errMsgs, err.Error())
+		}
+	}
+	if len(errMsgs) != 0 {
+		return fmt.Errorf(strings.Join(errMsgs, ","))
+	}
+	return nil
+}
+
+func (e *esGateway) Objects() (toCreate, toDelete []client.Object) {
+	toCreate = append(toCreate, e.esGatewaySecrets()...)
+	toCreate = append(toCreate, e.esGatewayService())
+	toCreate = append(toCreate, e.esGatewayRole())
+	toCreate = append(toCreate, e.esGatewayRoleBinding())
+	toCreate = append(toCreate, e.esGatewayServiceAccount())
+	toCreate = append(toCreate, e.esGatewayDeployment())
+	return toCreate, toDelete
+}
+
+func (e *esGateway) Ready() bool {
+	return true
+}
+
+func (e *esGateway) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (e esGateway) esGatewayRole() *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RoleName,
+			Namespace: render.ElasticsearchNamespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{""},
+				Resources:     []string{"secrets"},
+				ResourceNames: []string{},
```

Comment:
It would be a good practice to explicitly list the secrets you would like to watch, if possible.

Reply:
I'm not sure that will be possible given that each component will have 2 secrets each living in this namespace, and we can have any amount of managed cluster components connected, each of them also having 2 secrets per component.

Certainly for the management/standalone secrets we can explicitly list every components secret, but things get tricky with managed clusters since their secrets follow the format `component-name-cluster-name-elasticsearch-access-gateway` or `component-name-cluster-name-gateway-verification-credentials`. Is it worth looking into a way to list every connected managed cluster and to build a list of secrets that es-gateway is allowed to access explicitly? If so, maybe that can be a follow up task?